### PR TITLE
Fix return type for workspaces() in Girder4

### PR DESCRIPTION
**Pull request by eagw**
This change is necessary to make the sidebar listing of workspaces work in Girder4.

### CODE REVIEW

**Review by waxlamp — APPROVED** [review]
Congrats on filing your first PR! Now that this is approved, review your changes for any final things you might want to take care of, then hit the big green button!

**Comment by AlmightyYakob** [comment]
Hold off on merging please, I've got a suggestion or two.

File: src/index.ts [hunk 1]
```diff
@@ -50,6 +50,21 @@ export interface WorkspacePermissionsSpec {
...
+  modified: string;
+  arango_db_name: string;
+}
+
+export interface WorkspacesSpec {
+  count: number;
+  next: number;
+  previous: number;
+  results: Workspace[];
+}
```
Comment (AlmightyYakob) [inline src/index.ts:66]: I actually think having a generic interface for any response like this (which is very common in our api) is beneficial. 
```suggestion
export interface Paginated<T> {
  count: number,
  next: string,
  previous: string,
  results: T[],
}
```
I wouldn't actually commit this suggestion, instead I'd recommend just making this change locally, with the addition of replacing uses of `WorkspaceSpec` with `Paginated<Workspace>`.

**Review by AlmightyYakob — CHANGES_REQUESTED** [review]
Looks good, just one suggestion

File: src/index.ts (hunk 1 above)
Comment (eagw) [reply src/index.ts:66]: That makes sense- will do!

**Comment by waxlamp** [comment]
@eagw I've converted this to a draft PR so we can do a bit of further cleanup here. If you addressed Jake's [suggestion](https://github.com/multinet-app/multinetjs/pull/38#discussion_r672528399), then we should move those two extra commits to a different branch, then do a final review here and merge. Let's talk on Monday.

File: src/index.ts (hunk 1 above)
Comment (AlmightyYakob) [reply src/index.ts]: Looks like this is addressed by 27be0fa, thanks!

**Review by AlmightyYakob — APPROVED** [review]

**Pull request merged**